assets/src/scripts/components/graph/state/options.js

```diff
@@ -0,0 +1,88 @@
+import memoize from 'fast-memoize';
+import { createSelector } from 'reselect';
+
+const MOUNT_POINT = 'components/graph/options';
+const LITHOLOGY_VISIBILITY_SET = `${MOUNT_POINT}/LITHOLOGY_VISIBILITY_SET`;
+
+const DEFAULT_LITHOLOGY_VISIBILITY = true;
+
+
+/**
+ * Action creator to set the default visibility of the lithology layers in the
+ * water levels chart.
+ * @param {Object}  id      ID of component who visibility to set
+ * @param {Boolean} visible Visibility
+ */
+export const setLithologyVisibility = function (id, lithologyVisibility) {
+    return {
+        type: LITHOLOGY_VISIBILITY_SET,
+        payload: {
+            id,
+            lithologyVisibility
+        }
+    };
+};
+
+/**
+ * Returns a selector that:
+ * Returns the visibility of the lithology layers for a given component.
+ * If unset, the default visibility is loaded from localStorage, defaulting to true.
+ * @param  {Object} state Redux store
+ * @return {Object}       Visibility for given component
+ */
+export const getLithologyVisibility = memoize(opts => createSelector(
+    state => state.lithologyVisibility || {},
+    (lithologyVisibility) => {
+        // Always show the lithology layers on construction graphs
+        if (opts.graphType === 'construction') {
+            return true;
+        }
+
+        // If we don't have a visibility in the state, default to localStorage
+        if (lithologyVisibility[opts.id] === undefined) {
+            const visibility = window.localStorage.getItem('defaultLithologyVisibility');
```

Comment:
Not opposed to storing in the local storage but it is outside the scope of this ticket I think. Also the visibility in the localStorage is not stored by graph id as the visibility in the state is. Seems like this could cause problems if more than one chart shows lithology.